dotcom-rendering/src/model/enhanceCollections.ts

```diff
@@ -82,7 +99,8 @@ export const enhanceCollections = ({
 			{
 				canBeBranded:
 					!isOnPaidContentFront &&
-					collectionBranding?.kind === 'paid-content',
+					(collectionBranding?.kind === 'paid-content' ||
```

Comment:
What is `collectionBranding?.kind` equal to if the cards have different sponsors?

Reply:
I can see the comments in DCR codebase has `paid-content` is the same as `sponsored` so maybe a bit of clarification would be nice 🤓

Reply:
> What is `collectionBranding?.kind` equal to if the cards have different sponsors? Is it strange that it's not `paid-content`?

If they have different sponsors, `collectionBranding` is undefined, as per the logic in [decideCollectionBranding](https://github.com/guardian/dotcom-rendering/blob/4ea838cc9622a2748f3e7b90fe8f956d293e8e13/dotcom-rendering/src/lib/branding.ts#L186-L189). I'm interested in discussing whether this logic is correct. When I started updating it, it just raised further questions, as per PR description:

> I tried moving this check into `decideCollectionBranding`, but it's not clear that this is the correct place. The [`CollectionBranding`](https://github.com/guardian/dotcom-rendering/blob/77a303c606f6a7c625e99f301a78c21671242ece/dotcom-rendering/src/types/branding.ts#L34-L66) it returns has a mandatory [`branding`](https://github.com/guardian/dotcom-rendering/blob/77a303c606f6a7c625e99f301a78c21671242ece/dotcom-rendering/src/types/branding.ts#L53) property, which we wouldn't need in this case. Making `branding` optional seems illogical, given the type concerns itself with branding.

Reply:
> if we are not on paid content Front but still we want to have the Labs branding for paid content Labs ads even if it is for different sponsors? 

Yes, that's what this change is attempting to achieve.

> Does the same issue happen for Labs branding in paid content Front?

I _think_ this path is handled in [`decideCollectionBranding`](https://github.com/guardian/dotcom-rendering/blob/4ea838cc9622a2748f3e7b90fe8f956d293e8e13/dotcom-rendering/src/lib/branding.ts#L138-L151) (I could be wrong, I haven't tested)

Reply:
Thank you for the clarification! I think it would be nice to have WebX thoughts about where is the best place for this logic.

Reply:
> If they have different sponsors, `collectionBranding` is undefined, as per the logic in [decideCollectionBranding](https://github.com/guardian/dotcom-rendering/blob/4ea838cc9622a2748f3e7b90fe8f956d293e8e13/dotcom-rendering/src/lib/branding.ts#L186-L189). I'm interested in discussing whether this logic is correct. When I started updating it, it just raised further questions, as per PR description:
> 
> I tried moving this check into `decideCollectionBranding`, but it's not clear that this is the correct place. The [`CollectionBranding`](https://github.com/guardian/dotcom-rendering/blob/77a303c606f6a7c625e99f301a78c21671242ece/dotcom-rendering/src/types/branding.ts#L34-L66) it returns has a mandatory [`branding`](https://github.com/guardian/dotcom-rendering/blob/77a303c606f6a7c625e99f301a78c21671242ece/dotcom-rendering/src/types/branding.ts#L53) property, which we wouldn't need in this case. Making `branding` optional seems illogical, given the type concerns itself with branding.

I think this would be nicer to remove the `everyCardHasSameSponsor` (if `kind === 'paid-content'` perhaps) check in https://github.com/guardian/dotcom-rendering/blob/4ea838cc9622a2748f3e7b90fe8f956d293e8e13/dotcom-rendering/src/lib/branding.ts#L186-L189

And have `branding` undefined.

`CollectionBranding` does allow branding to be undefined if `kind === 'editorial`, so I would say that's fine if it's optional for `paid-content`.

Reply:
Thanks for the suggestion @Jakeii, I would prefer this logic to live entirely in decideCollectionBranding, so I'm on board. I have [tried it to a fashion](https://github.com/guardian/dotcom-rendering/compare/main...sa-display-paid-content-palette-2), and it's almost there. The logos are no longer appearing on the cards, which we'd need to fix.

<img width="1319" alt="Screenshot 2023-12-21 at 12 01 23" src="https://github.com/guardian/dotcom-rendering/assets/5931528/fe2c22f5-327b-43d5-83a6-1d26c7bc3215">